.github/workflows/ossf-scorecard.yml

```diff
@@ -38,7 +38,7 @@ jobs:
           persist-credentials: false
 
       - name: "Run analysis"
-        uses: ossf/scorecard-action@0864cf19026789058feabb7e87baa5f140aac736  # v2.3.1
+        uses: ossf/scorecard-action@62b2cac7ed8198b15735ed49ab1e5cf35480ba46  # v2.4.0
```

Comment:
This is a great example of why I dislike this "pinning".  We fail to get updates, bugfix or otherwise without having to manually update these pins.  This is going to be a constant churn and we are going to be vulnerable to bugs and security issues (I.e. imagine that there was a security issue in `ossf/scorecard` _v2.3.1_ that was fixed in something later than that version -- we are vulnerable to that security bug until somebody becomes aware of it and updates this repo and all of the others we use this in to pin a new version -- who is responsible for knowing that _v2.3.1_ of `ossf/scorecard` has a security vulnerability?  multiply that kind of awareness requirement across all of the number of dependencies that we are being expected to pin and see how unmanageable that is in reality) that are fixed in later versions after the one we have pinned rather than automatically getting them by using a tag.

Reply:
I don't like it either.
I think the only solution now is to have a bot that monitors all actions and checks if pined dependencies are up-to-date.
Have to take a look at Dependabot: https://josh-ops.com/posts/github-dependabot-for-actions/

Reply:
FWIW I agree. Pinning hashes really sucks